src/redux/slices/workspace-slice.ts

```diff
@@ -285,6 +340,190 @@ const workspacesSlice = createSlice({
                 });
             }
         },
+
+        // ==================== Panel Association Operations ====================
+        associateSldToNad: (state, action: PayloadAction<{ sldPanelId: UUID; nadPanelId: UUID }>) => {
+            const { sldPanelId, nadPanelId } = action.payload;
+            const workspace = getActiveWorkspace(state);
+
+            // Update NAD panel metadata to include SLD in associated list (prevent duplicates)
+            updatePanel(state, nadPanelId, (panel) => {
+                const metadata = panel.metadata as NADPanelMetadata;
+                const associatedPanels = metadata.associatedVoltageLevelPanels || [];
+                // Only add if not already associated
+                if (!associatedPanels.includes(sldPanelId)) {
+                    panel.metadata = {
+                        ...metadata,
+                        associatedVoltageLevelPanels: [...associatedPanels, sldPanelId],
+                    };
+                }
+            });
+
+            // Keep focus on NAD panel (bring to front and un-minimize if needed)
+            bringToFront(workspace, nadPanelId);
+        },
+
+        dissociateSldFromNad: (state, action: PayloadAction<UUID>) => {
+            const sldPanelId = action.payload;
+            const workspace = getActiveWorkspace(state);
+            const nadPanelId = findNadForSld(workspace, sldPanelId);
+
+            // Remove from NAD panel's associated list
+            if (nadPanelId && workspace.panels[nadPanelId]) {
+                updatePanel(state, nadPanelId, (panel) => {
+                    const nadMetadata = panel.metadata as NADPanelMetadata;
+                    panel.metadata = {
+                        ...nadMetadata,
+                        associatedVoltageLevelPanels: (nadMetadata.associatedVoltageLevelPanels || []).filter(
+                            (id) => id !== sldPanelId
+                        ),
+                    };
+                });
+            }
+
+            // Update orderIndex to place at end of dock
+            updatePanel(state, sldPanelId, (panel) => {
+                panel.orderIndex = workspace.nextOrderIndex++;
+            });
+
+            // Bring the dissociated SLD panel to front
+            bringToFront(workspace, sldPanelId);
+        },
+
+        createNadAndAssociateSld: (
+            state,
+            action: PayloadAction<{ sldPanelId: UUID; voltageLevelId: string; voltageLevelName: string }>
+        ) => {
+            const { sldPanelId, voltageLevelId, voltageLevelName } = action.payload;
+            const workspace = getActiveWorkspace(state);
+            const sldPanel = workspace.panels[sldPanelId];
+
+            // Create NAD panel with the voltage level, using SLD's position and size
+            createPanel(workspace, PanelType.NAD, {
+                title: voltageLevelName,
+                metadata: {
+                    initialVoltageLevelIds: [voltageLevelId],
```

Comment:
I did not check deeply if it was possible, but could you instead create the panel containing the VL sld (with panelId = "vlPanelId", then create the panel containing the NAD with associatedVoltageLevelPanels = [vlPanelId] ?

It would prevent using this "hack"